addons/event_sale/tests/test_event_sale.py

```diff
@@ -121,3 +121,50 @@ def test_01_ticket_price_with_pricelist_and_tax(self):
         })
         sol.product_id_change()
         self.assertEqual(so.amount_total, 660.0, "Ticket is $1000 but the event product is on a pricelist 10 -> 6. So, $600 + a 10% tax.")
+
+    def test_02_delete_associated_sale_order(self):
+        event_product = self.env['product.template'].create({
```

Comment:
Why not reusing existing data done in setup ?

Reply:
I updated the tests. As discussed, I also added an `ondelete` attribute to the field `registration_editor.sale_order_id` (commit message is also updated)